pkg/test/framework/suite_test.go

```diff
@@ -450,6 +450,49 @@ func TestSuite_DoubleInit_Error(t *testing.T) {
 	g.Expect(errCode2).NotTo(Equal(0))
 }
 
+func TestSuite_GetResource(t *testing.T) {
+	defer cleanupRT()
+	g := NewGomegaWithT(t)
+
+	var (
+		structRes    fakeEnvironment
+		interfaceRes resource.Cluster
+
+		structErr, ifErr, sliceErr, nonPtrErr error
+	)
+	sliceRes := []resource.Cluster{fakeCluster{index: 3}}
+
+	runFn := func(ctx *suiteContext) int {
+		structErr = ctx.GetResource(&structRes)
+		ifErr = ctx.GetResource(&interfaceRes)
+		sliceErr = ctx.GetResource(&sliceRes)
+		nonPtrErr = ctx.GetResource(fakeEnvironment{})
+		return 0
+	}
+
+	s := newSuite("tid", runFn, defaultExitFn, defaultSettingsFn)
+	s.Setup(func(c resource.Context) error {
+		c.TrackResource(fakeEnvironment{numClusters: 2})
+		c.TrackResource(fakeCluster{index: 1})
+		return nil
+	})
+	s.Run()
+
+	g.Expect(structErr).To(BeNil())
```

Comment:
Can we change this to be table-based, using subtests?

Reply:
So at first I put it in the `map[string]struct{ /* test case field */ }` structure but the fact that my `ref` field had type `interface` confused the reflection a bit. There is a workaround where I can do 

```
ref: func() interface{} {
   ref := &fakeCluster
   return &ref
}
```

to ensure the test case preserves type information and treats `tt.ref` as `**fakeCluster` rather than `*interface{}` but then things get ugly and I have to bring in reflection just to do assertions.

I broke things into subtests with a common `act` function, I think it's much more readable.